docs/GettingStarted.md

```diff
@@ -355,6 +357,58 @@ Where `options` is an optional `Hash` that accepts the following parameters:
 | ``service_name`` | Service name used for `elasticsearch` instrumentation | elasticsearch |
 | ``quantize`` | Hash containing options for quantization. May include `:show` with an Array of keys to not quantize (or `:all` to skip quantization), or `:exclude` with Array of keys to exclude entirely. | {} |
 
+### Excon
+
+The `excon` integration is available through the `ddtrace` middleware:
+
+```ruby
+require 'excon'
+require 'ddtrace'
+
+# Configure default Excon tracing behavior
+Datadog.configure do |c|
+  c.use :excon, service_name: 'excon'
+end
+
+connection = Excon.new('https://example.com')
+connection.get
+```
+
+Where `options` is an optional `Hash` that accepts the following parameters:
+
+| Key | Description | Default |
+| --- | --- | --- |
+| `service_name` | Service name for Excon instrumentation. When provided to middleware for a specific connection, it applies only to that connection object. | `'excon'` |
+| `split_by_domain` | Uses the request domain as the service name when set to `true`. | `false` |
+| `distributed_tracing` | Enables [distributed tracing](#distributed-tracing) | `false` |
+| `error_handler` | A `Proc` that accepts a `response` parameter. If it evaluates to a *truthy* value, the trace span is marked as an error. By default only sets 5XX responses as errors. | `nil` |
+| `tracer` | A `Datadog::Tracer` instance used to instrument the application. Usually you don't need to set that. | `Datadog.tracer` |
+
+**Configuring connections to use different settings**
```

Comment:
In other integrations, we may do something like:
```ruby
Datadog.configure(client, service_name: 'something_else')
```

I guess we can't do it for `Excon` (and so the middleware must be added manually) because once initialized it cannot be changed? Consider it as a comment, if it introduces too much complexity we can keep it that way.

Reply:
@palazzem Right. If you look at Faraday, which is almost identical to Excon, we can't do this either. Both of these integrations implement a middleware scheme, which require middleware to be injected when the connection/client object is created.

It might be possible to change the configuration to be derived from a `Pin` (much like the Sequel integration), which would permit the `Datadog.configure` strategy to work for a limited set of options (e.g. `service_name`.) But that would also introduce monkey-patching into a library that already provides an API for instrumentation. Not to say that might not be worth it, but it would represent an larger overhaul that does run contrary to the gem author's intentions.

I'd recommend keeping this as is for now, but scheduling a task to refactor it to support that `Datadog.configure` strategy in the future.

Reply:
Agreed. Thanks!